"Talosctl config" segfaults when config points to non-existing context

### Description
I am playing around with CAPI, and as such, am using the context option in the config file a lot. Turned out, when you set the `context` key to something that's not present in `contexts`, the `config` subcommand segfaults.

Other subcommands (such as `version`) are aware that the context is wrong, however.

```
context: doesnotexist
contexts:
  target:
    endpoints:
      - 10.8.95.95
    nodes:
      - 10.8.95.95
    ca: 1234
    crt: 1234
    key: 1234
```

### Logs
```
$ talosctl config node 1.2.3.4
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1a920ac]

goroutine 1 [running]:
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.glob..func4(0x2eebb40, 0xc000762890, 0x1, 0x1, 0x0, 0x0)
	/src/cmd/talosctl/cmd/talos/config.go:72 +0x10c
github.com/spf13/cobra.(*Command).execute(0x2eebb40, 0xc000762870, 0x1, 0x1, 0x2eebb40, 0xc000762870)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x2ee7ee0, 0x2f06b50, 0x1f32db2, 0x9)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/talos-systems/talos/cmd/talosctl/cmd.Execute(0x0, 0x0)
	/src/cmd/talosctl/cmd/root.go:42 +0x235
main.main()
	/src/cmd/talosctl/main.go:18 +0x33
```

### Environment
- Talos version: 0.7.0
- Kubernetes version: n/a
- Platform: amd64

--- a/cmd/talosctl/cmd/talos/config.go
+++ b/cmd/talosctl/cmd/talos/config.go
@@ -36,12 +36,9 @@ var configEndpointCmd = &cobra.Command{
 	Long:    ``,
 	Args:    cobra.MinimumNArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		c, err := clientconfig.Open(Talosconfig)
+		c, err := openConfigAndContext("")
 		if err != nil {
-			return fmt.Errorf("error reading config: %w", err)
-		}
-		if c.Context == "" {
-			return fmt.Errorf("no context is set")
+			return err
 		}
 
 		c.Contexts[c.Context].Endpoints = args
@@ -61,12 +58,9 @@ var configNodeCmd = &cobra.Command{
 	Long:    ``,
 	Args:    cobra.ArbitraryArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		c, err := clientconfig.Open(Talosconfig)
+		c, err := openConfigAndContext("")
 		if err != nil {
-			return fmt.Errorf("error reading config: %w", err)
-		}
-		if c.Context == "" {
-			return fmt.Errorf("no context is set")
+			return err
 		}
 
 		c.Contexts[c.Context].Nodes = args
@@ -87,9 +81,9 @@ var configContextCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) error {
 		context := args[0]
 
-		c, err := clientconfig.Open(Talosconfig)
+		c, err := openConfigAndContext(context)
 		if err != nil {
-			return fmt.Errorf("error reading config: %w", err)
+			return err
 		}
 
 		c.Context = context
@@ -158,6 +152,27 @@ var configGenerateCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("'talosctl config generate' was renamed to 'talosctl gen config'")
 	},
+}
+
+func openConfigAndContext(context string) (*clientconfig.Config, error) {
+	c, err := clientconfig.Open(Talosconfig)
+	if err != nil {
+		return nil, fmt.Errorf("error reading config: %w", err)
+	}
+
+	if context == "" {
+		context = c.Context
+	}
+
+	if context == "" {
+		return nil, fmt.Errorf("no context is set")
+	}
+
+	if _, ok := c.Contexts[context]; !ok {
+		return nil, fmt.Errorf("context %q is not defined", context)
+	}
+
+	return c, nil
 }
 
 func init() {

